Added method for dynamic font loading

krista-koivisto opened the pull request:
Added two new methods to the template:

1. `GoogleFonts.asMap()` which returns a map where the key is the font family and the value is the corresponding `GoogleFonts` method.
2. `GoogleFonts.getFont(String fontFamily, {...})` which looks up the method by font family name using `GoogleFonts.asMap()`, runs it and returns the resulting `TextStyle`. The {...} "parameter" here represents the standard parameters for font methods, I decided to omit them for clarity.

Also, added a unit test which tests that `GoogleFonts.getFont` returns the same `TextStyle` instance as the regular method call does.

These additions were made in response to #34.

krista-koivisto commented:
@googlebot I signed it!

clocksmith commented on example/lib/main.dart:
```diff
@@ -47,7 +47,7 @@ class _MyHomePageState extends State<MyHomePage> {
             Text(
               'You have pushed the button this many times:',
-              style: GoogleFonts.oswald(textStyle: display1),
+              style: GoogleFonts.getFont("Oswald", textStyle: display1),
```
nit: use single quotes for strings

clocksmith replied:
Actually, I don't think this should be changed here. We would prefer users to use `GoogleFonts.oswald` over `GoogleFonts.getFont('Oswald'` unless that had to use the string based API

clocksmith commented on test/generated_font_methods_test.dart:
```diff
@@ -659,4 +659,15 @@ main() {
+  //////////////////
+
+  testWidgets('getFont returns the correct font', (tester) async {
+    final dynamicFont = GoogleFonts.getFont("Roboto Mono");
```
nit: single quotes for string

clocksmith commented on generator/google_fonts.tmpl:
```diff
@@ -29,6 +29,94 @@ class GoogleFonts {
+    double decorationThickness,
+  })> asMap() => const {
+    {{#method}}
+    "{{fontFamilyDisplay}}": GoogleFonts.{{methodName}},
```
Single quotes can also be used here

krista-koivisto commented:
Thank you, @clocksmith!

Applied requested changes, as well as:

 * Changed example in `README.md` to use single quotes.
 * Removed redundant word in comment for `getFont`.
